### PR TITLE
feat(cmp): wrap lisp functions instead of appending paran

### LINKED ISSUE #175: Wrap lisp function instead of appending `()`
**Is your feature request related to a problem? Please describe.**

I find my self disabling the simple yet awesome cmp addition, as you may know, in lisp functions get wrapped wtih parentheses instead of having `()`  appended it to it.

**Describe the solution you'd like**

Wrap function names with () in lisp filetypes `{ "clojure", "clojurescript", "fennel", "janet", ........}`.

### PR DESCRIPTION
**Pull request by tami5**
### Purpose

closes #175

- [x] make lisp filetype configurable? 
  - opt 1: through appending/redfining 'nvim-autopairs.completion.cmp'.lisp?
  - opt 2: pass extra lisp filetypes in on confirm function?

### CODE REVIEW

**Comment by windwp** [comment]
LGTM 
Can you add some information on Readme.md ?

**Comment by tami5** [comment]
> LGTM Can you add some information on Readme.md ?

hmmm, but which configuration option should we choose. Also, you want something like support for lisp functions in cmp section?

**Comment by windwp** [comment]
i think we can still disable it by  
```lua
map_char={
  fennel=''
}
-- and they can update it by update
require('nvim-autopairs.completion.cmp').lisp
```
it you want to change config by on_confirm_done function it will be fine it is your choice :)

what do you mean about lisp functions?

**Comment by tami5** [comment]
> what do you mean about lisp functions?

Like saying, cmp both support appending `()` and wrapping lisp functions.

There's an issue I faced while testing, when inside `()` it should avoid wrapping, I'm going to try to fix this later this evening

File: lua/nvim-autopairs/completion/cmp.lua [hunk 1]
```diff
@@ -3,49 +3,57 @@ local utils = require('nvim-autopairs.utils')
...
-                end
-                vim.api.nvim_feedkeys(char, 'i', true)
-            end
+        if ignore_append(char, opt.kinds, next_char, prev_char, item)  then
+            return
+        end
+
+        if utils.is_in_table(M.lisp, vim.bo.filetype) then
+            local length = #item.label
+            utils.feed(utils.key.left, length)
```
Comment (tami5) [inline lua/nvim-autopairs/completion/cmp.lua:51]: maybe check if `length - 1 char is `(`` and do nothing if it is

**Comment by tami5** [comment]
Okay @windwp I believe now it should be merge, unless you have additional thoughts.

**Comment by windwp** [comment]
ok thank .

**Pull request merged**